apm-agent-core/src/main/java/co/elastic/apm/agent/util/PackageScanner.java

```diff
@@ -91,7 +91,7 @@
             @Override
             public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
                 if (file.toString().endsWith(".class")) {
-                    classNames.add(basePackage + "." + basePath.relativize(file).toString().replace(File.separatorChar, '.').replace(".class", ""));
+                    classNames.add(basePackage + "." + basePath.relativize(file).toString().replace(File.separatorChar, '.').replace('/', '.').replace(".class", ""));
```

Comment:
Excellent. I would add a comment, so better add a variable for that. Something like:
```suggestion
                    // We need to escape both the filesystem-specific separator and the explicit `/` separator that may be added by the relativize() implementation
                    String classNameSuffix = basePath.relativize(file).toString().replace(File.separatorChar, '.').replace('/', '.').replace(".class", "");
                    classNames.add(basePackage + "." + classNameSuffix);
```